stdlib/3/fileinput.pyi

```diff
@@ -0,0 +1,49 @@
+from typing import Iterable, BinaryIO, List
+
+
+def input(
+    files: List[str]=...,
+    inplace: bool=...,
+    backup: str=...,
+    bufsize: int=...,
+    mode: str=...,
+    openhook: Callable[[str, str], BinaryIO]=...
```

Comment:
I think all occurrences of BinaryIO need to be changed to IO[AnyStr]. The module appears to work for binary as well as text files, depending on the value of the mode argument. The return value of `__next__()` and other methods that return data from the file need to be AnyStr as well. The PEP 484 type system currently cannot express this properly. I think the FileInput class needs to be explicitly inheriting from Iterable[AnyStr] too, and you'd have to add Generic[AnyStr] too I think (please write some test files yourself to verify these edge cases).